### PR TITLE
desupport Ubuntu/trusty for nightlies

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
following #296

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_deb_coreproject.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@
           values:
           - debian
     execution-strategy:
-      combination-filter: '((onlyos == "all") || onlyos == os) && !(arch == "armv7" && (os == "trusty")) && !(arch == "armv8" && os == "trusty") && !((version == "1.14" || version == "1.15") && os == "stretch")'
+      combination-filter: '((onlyos == "all") || onlyos == os) && !(os == "trusty" && repo == "develop") && !(arch == "armv7" && (os == "trusty")) && !(arch == "armv8" && os == "trusty") && !((version == "1.14" || version == "1.15") && os == "stretch")'
```
Comment (domcleal) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_deb_coreproject.yaml]: 1. Prefer testing `version` as repo may vary when building against user branches in PRs etc.
2. Prefer listing which releases it should build for (realistically, only 1.15 now), then it won't need updating later.

(Copy the Stretch conditional.)

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_deb_dependency.yaml [hunk 2]
```diff
@@ -35,7 +35,7 @@
           values:
           - debian
     execution-strategy:
-      combination-filter: '((onlyos == "all") || onlyos == os) && !(arch == "armv7" && (os == "trusty")) && !(arch == "armv8" && os == "trusty")'
+      combination-filter: '((onlyos == "all") || onlyos == os) && !(os == "trusty" && repo == "develop") && !(arch == "armv7" && (os == "trusty")) && !(arch == "armv8" && os == "trusty")'
```
Comment (domcleal) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_deb_dependency.yaml]: Ditto, use version.

**Comment by mmoll** [comment]
updated.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/release_push_deb.yaml [hunk 3]
```diff
@@ -24,4 +24,4 @@
           - trusty
           - xenial
     execution-strategy:
-      combination-filter: 'os == "jessie" || (os == "stretch" && major_version != "1.14" && major_version != "1.15") || os == "trusty" || os == "xenial"'
+      combination-filter: 'os == "jessie" || (os == "stretch" && major_version != "1.14" && major_version != "1.15") || (os == "trusty" && major_version="1.15") || os == "xenial"'
```
Comment (domcleal) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/release_push_deb.yaml]: Use `==`, not single `=`.

Comment (mmoll) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/release_push_deb.yaml]: fixed.

**Comment by domcleal** [comment]
Merged, thanks @mmoll.

**Pull request merged**